npm/npm-debug-cli/dataplaneConverter/converter.go

```diff
@@ -0,0 +1,300 @@
+		EmitUnpopulated: true,
+	}
+	if len(filenames) > 0 {
+		pbRuleObj = c.GetProtobufRulesFromIptable(tableName, filenames[0], filenames[1])
+	} else {
+		pbRuleObj = c.GetProtobufRulesFromIptable(tableName)
+	}
+	for _, rule := range pbRuleObj {
+		ruleJson, err := m.Marshal(rule) // pretty print
+		if err != nil {
+			log.Fatalf("Error occured during marshaling. Error: %s", err.Error())
+		}
+		ruleResListJson = append(ruleResListJson, ruleJson)
+	}
+	return ruleResListJson
+
+}
+
+// GetRulesFromIptable returns a list of protobuf rule object of an iptable. Can pass in npmCache file and iptable-save files in that order for debugging purposes.
+func (c *Converter) GetProtobufRulesFromIptable(tableName string, filenames ...string) []*pb.RuleResponse {
+	p := &parser.Parser{}
+	var ipTableObj *iptable.Iptables
+
+	if len(filenames) > 0 {
+		c.initConverter(filenames[0])
+		ipTableObj = p.ParseIptablesObject(tableName, filenames[1])
+
+	} else {
+		c.initConverter()
+		ipTableObj = p.ParseIptablesObject(tableName)
+	}
+	ruleResList := make([]*pb.RuleResponse, 0)
+	for _, v := range ipTableObj.Chains() {
+		chainRules := c.getRulesFromChain(v)
+		ruleResList = append(ruleResList, chainRules...)
+	}
+
+	return ruleResList
+
+}
+
+func (c *Converter) getRulesFromChain(iptableChainObj *iptable.IptablesChain) []*pb.RuleResponse {
+	rules := make([]*pb.RuleResponse, 0)
+	for _, v := range iptableChainObj.Rules() {
+		rule := &pb.RuleResponse{}
+		rule.Chain = iptableChainObj.Name()
+		if _, ok := c.RequiredChainsMap[rule.Chain]; !ok {
+			continue
+		}
+		rule.Protocol = v.Protocol()
+		switch v.Target().Name() {
+		case "MARK":
+			rule.Allowed = true
+		case "DROP":
+			rule.Allowed = false
+		default:
+			// ignore other targets
+			continue
+		}
+		direction := c.getRuleDirection(iptableChainObj.Name())
+		if direction >= 0 {
+			rule.Direction = direction
+		}
+
+		c.getModulesFromRule(v.Modules(), rule)
+		rules = append(rules, rule)
+
+	}
+	return rules
+}
+
+func (c *Converter) getRuleDirection(iptableChainObjName string) pb.Direction {
+	if strings.Contains(iptableChainObjName, "EGRESS") {
+		return pb.Direction_EGRESS
+	} else if strings.Contains(iptableChainObjName, "INGRESS") {
+		return pb.Direction_INGRESS
+	} else {
+		return -1
+	}
+}
+
+func (c *Converter) getSetType(name string, m string) pb.SetType {
+	if m == "ListMap" { // labels of namespace
+		if strings.Contains(name, util.IpsetLabelDelimter) {
+			if strings.Count(name, util.IpsetLabelDelimter) > 1 {
+				return pb.SetType_NESTEDLABELOFPOD
+			}
+			return pb.SetType_KEYVALUELABELOFNAMESPACE
+		}
+		return pb.SetType_KEYLABELOFNAMESPACE
+	} else {
+		if strings.HasPrefix(name, util.NamespacePrefix) {
+			return pb.SetType_NAMESPACE
+		}
+		if strings.HasPrefix(name, util.NamedPortIPSetPrefix) {
+			return pb.SetType_NAMEDPORTS
+		}
+		if strings.Contains(name, util.IpsetLabelDelimter) {
+			return pb.SetType_KEYVALUELABELOFPOD
+		}
+		return pb.SetType_KEYLABELOFPOD
+	}
```

Comment:
these nest pretty deep, can you simplify since you're returning at each leaf?
you can drop the `else` entirely and outdent all of those checks, because you always return at the end of your `if` branch